lib/orchestrator/protocol_handler.ex

```diff
@@ -96,19 +96,19 @@ defmodule Orchestrator.ProtocolHandler do
   def handle_message(pid, monitor_logical_name, message) do
     case Integer.parse(message) do
       {len, rest} ->
-        message_body = String.slice(rest, 1, len)
-        if (len > String.length(message_body)) do
-          # Incomplete message
-          # Send message back if we can't process it
-          # so that future data can be appended
-          {:incomplete, message}
-        else
-          GenServer.cast(pid, {:message, message_body})
-          # If there's more, try to process more (but it may be incomplete)
-          handle_message(pid, monitor_logical_name, String.slice(rest, 1 + len, 100_000))
+        case rest do
+          <<" ", message_body::binary-size(len), new_rest::binary>> ->
+            GenServer.cast(pid, {:message, message_body})
+            # If there's more, try to process more (but it may be incomplete)
+            handle_message(pid, monitor_logical_name, new_rest)
+        _ ->
```

Comment:
Stylistic nit - I like to keep variables as local as possible, so I'd write `incomplete -> {:incomplete, incomplete}`. Probably mostly a matter of preference.